src/AuthContext.tsx

```diff
@@ -68,17 +77,36 @@ export function AuthProvider({ children }: { children: React.ReactNode }) {
   useEffect(() => {
     const unsubscribe = onAuthStateChanged(auth, (user) => {
       setCurrentUser(user);
+
+      if (user == null) {
+        setMongoUser(null);
+      } else {
+        user?.getIdToken().then((t) => {
+          if (t == null) {
+            console.log("No token available");
+          } else {
+            //review: should there be error handling here and how would it work (error state in context maybe?)
```

Comment:
im thinking that if you cant find the info, display an error message and then logout the user from firebase. I'm gonna talk to berke about that